client/ayon_core/hosts/zbrush/api/lib.py

```diff
@@ -0,0 +1,112 @@
+from ayon_core.pipeline.template_data import get_template_data
+
+
+log = logging.getLogger("zbrush.lib")
+
+
+def execute_zscript(zscript, communicator=None):
+    if not communicator:
+        communicator = CommunicationWrapper.communicator
+    return communicator.execute_zscript(zscript)
+
+
+def find_first_filled_path(path):
+    if not path:
+        return ""
+
+    fields = set()
+    for item in Formatter().parse(path):
+        _, field_name, format_spec, conversion = item
+        if not field_name:
+            continue
+        conversion = "!{}".format(conversion) if conversion else ""
+        format_spec = ":{}".format(format_spec) if format_spec else ""
+        orig_key = "{{{}{}{}}}".format(
+            field_name, conversion, format_spec)
+        fields.add(orig_key)
+
+    for field in fields:
+        path = path.split(field, 1)[0]
+    return path
+
+
+def get_workdir(project_name, asset_name, task_name):
+    project = get_project(project_name)
+    asset = get_asset_by_name(project_name, asset_name)
+
+    data = get_template_data(project, asset, task_name)
+
+    anatomy = Anatomy(project_name)
+    workdir = anatomy.templates_obj["work"]["folder"].format(data)
+
+    # Remove any potential un-formatted parts of the path
+    valid_workdir = find_first_filled_path(workdir)
+
+    # Path is not filled at all
+    if not valid_workdir:
+        raise AssertionError("Failed to calculate workdir.")
+
+    # Normalize
+    valid_workdir = os.path.normpath(valid_workdir)
+    if os.path.exists(valid_workdir):
+        return valid_workdir
+
+    data.pop("task", None)
+    workdir = anatomy.templates_obj["work"]["folder"].format(data)
+    valid_workdir = find_first_filled_path(workdir)
+    if valid_workdir:
+        # Normalize
+        valid_workdir = os.path.normpath(valid_workdir)
+        if os.path.exists(valid_workdir):
+            return valid_workdir
+
+
+def get_current_file():
+    host = registered_host()
+    return host.get_current_workfile()
+
+
+def execute_publish_model(filepath):
+    save_file_zscript = ("""
+[IFreeze,
+[VarSet, filepath, "{filepath}"]
+[FileNameSetNext, #filepath]
+[IKeyPress, 13, [IPress, Tool:Export:Export]]]
+]
+""").format(filepath=filepath)
+    execute_zscript(save_file_zscript)
+
+
+def has_edit_mode():
+    tmp_output_file = tempfile.NamedTemporaryFile(
+        mode="w", prefix="a_zb_", suffix=".txt", delete=False
+    )
+    tmp_output_file.close()
+    temp_file = tmp_output_file.name.replace("\\", "/")
+
+    in_edit_mode = ("""
+[IFreeze,
+[MemCreate, EditMode, 20, 0]
+[VarSet, InEditMode, [IsEnabled, Transform:Edit]]
+[Note, InEditMode]
+[MemWriteString, EditMode, #InEditMode, 0]
+[MemSaveToFile, EditMode, "{temp_file}", 0]
+[MemDelete, EditMode]
+]
+""").format(temp_file=temp_file)
+    execute_zscript(in_edit_mode)
+    with open(temp_file) as mode:
+        content = str(mode.read())
+        bool_mode = content.rstrip('\x00')
+        return bool_mode
```

Comment:
Would this be possible to write in a way that we're within the context manager when querying the resulting value so that the actual temp file gets deleted?

Or do we need it closed because otherwise zbrush can't write into it? or?

Reply:
Maybe we can do not only for edit_mode(validation) but the export model function(extraction).

Reply:
Yup, I think the logic can basically be shared across all functions where you need to have a return value via a file.

_It would be even better if we could somehow do this without needing the file - but that's optimizations mostly and heavily dependent on what's actually possible from Zbrush scripts - this is likely very limited by their API._